source/FreeRTOS_IP_Utils.c

```diff
@@ -904,21 +917,18 @@ uint16_t usGenerateChecksum( uint16_t usSum,
     xSum.u32 = ( uint32_t ) xSum.u16[ 0 ] + xSum.u16[ 1 ] + ulCarry;
 
     uxDataLengthBytes %= 16U;
-    xLastSource.u8ptr = ( uint8_t * ) ( xSource.u8ptr + ( uxDataLengthBytes & ~( ( size_t ) 1 ) ) );
 
     /* Half-word aligned. */
+    ulSize = ( ( uxDataLengthBytes & ~( ( size_t ) 1U ) ) );
 
-    /* Coverity does not like Unions. Warning issued here: "The operator "<"
-     * is being applied to the pointers "xSource.u16ptr" and "xLastSource.u16ptr",
-     * which do not point into the same object." */
-    while( xSource.u16ptr < xLastSource.u16ptr )
+    for( ulX = 0U; ulX < ulSize; ulX += 2U )
```

Comment:
Please check if we can have ulX += 1U * sizeof( uint16_t ) , instead of of ulX += 2U at Line 924.